src/java/fr/paris/lutece/plugins/forms/web/FormXPage.java

```diff
@@ -666,6 +679,20 @@ public synchronized XPage doSaveFormResponse( HttpServletRequest request ) throw
                 return getStepView(  request );
             }
             FormsResponseUtils.fillResponseManagerWithResponses( request, true, _formResponseManager, _stepDisplayTree.getQuestions( ), false );
+
+            List<Step> stepList = StepHome.getStepsListByForm( form.getId( ) );
```

Comment:
We should use FormsCacheService to avoid performance problems (and keep consistency during the process)